docs/libraries/libraries-with-cli.md

```diff
@@ -0,0 +1,556 @@
+                "System.Text.RegularExpressions": "4.0.11-beta-23409",
+                "System.Threading.Tasks": "4.0.11-beta-23409"
+            }
+        }
+    }
+}
+
+```
+
+Note that framework assemblies being used are explicitly referenced in the `net40` target, and NuGet references are also explictly listed in the `dotnet55` target.
+
+Next, your `using`s in your source file can be adjusted like this:
+
+```csharp
+#if NET40
+// This only compiles for non-.NET 4.0 targets
+using System.Net;
+#else
+// This compiles for all other targets
+using System.Net.Http;
+using System.Threading.Tasks;
+#endif
+```
+
+The compiler is aware of the following names:
+
+* `NET20` -->  .NET Framework 2.0
+* `NET35` -->  .NET Framework 3.5
+* `NET40` -->  .NET Frameowrk 4.0
+* `NET45` -->  .NET Framework 4.5
+* `NET451` --> .NET Framework 4.5.1
+* `NET452` --> .NET Framework 4.5.2
+* `NET46` --> .NET Framework 4.6
+
+And further down in the source, you can use guards to use those libraries conditionally:
+
+```csharp
+    public class Library
+    {
+#if NET40
+        private readonly WebClient _client = new WebClient();
+        private readonly object _locker = new object();
+#else
+        private readonly HttpClient _client = new HttpClient();
+#endif
+
+#if NET40
+        // .NET 4.0 does not have async/await
+        public string GetDotNetCount()
+        {
+            string url = "http://www.dotnetfoundation.org/";
+          
+            var uri = new Uri(url);
+            
+            string result = "";
+            
+            // Lock here to provide thread-safety.
+            lock(_locker)
+            {
+                result = _client.DownloadString(uri);
+            }
+            
+            int dotNetCount = Regex.Matches(result, ".NET").Count;
+            
+            return $"Dotnet Foundation mentions .NET {dotNetCount} times!";
+        }
+#else
+        // .NET 4.5+ can use async/await!
+        public async Task<string> GetDotNetCountAsync()
+        {
+            string url = "http://www.dotnetfoundation.org/";
+            
+            // HttpClient is thread-safe, so no need to explicitly lock here
+            var result = await _client.GetStringAsync(url);
+            
+            int dotNetCount = Regex.Matches(result, ".NET").Count;
+            
+            return $"dotnetfoundation.orgmentions .NET {dotNetCount} times in its HTML!";
+        }
+#endif
+    }
+```
+
+And that's it!
+
+### But What about Portable Class Libraries (PCLs)?
+
+PCLs add one more thing to do before you can use `#if`s to conditionally compile different targets: **you need to add a compilation definition in your** `project.json` **file**.
+
+For example, if you wanted to target PCL profile 328 (.NET 4.0, Windows 8, Windows Phone Silverlight 8, Windows Phone 8.1, Silverlight 5.0), you may want to refer to it to as "PORTABLE328" when cross-compiling.  Simply add it to the `project.json` file as a `compilationOptions` attribute:
```

Comment:
should you link to that profiles page again here when you mention profile 328? Like "if you want to target [PCL profile 328] (http://embed.plnkr.co/03ck2dCtnJogBKHJ9EjY/preview)"